File: llm/lib/tasks/classify-page.mjs
// classify-page — classify a page into entity/concept/synthesis or none.
import { runJsonPrompt } from '../runner.mjs';

const VALID = new Set(['source', 'entity', 'concept', 'synthesis']);

export async function run({ kbRoot, input }) {
  const pagePath = input?.page_path || '';
  const title = input?.title || '';
  const body = input?.body || '';
  const { data } = await runJsonPrompt(kbRoot, 'classify-page', { page_path: pagePath, title, body });
  const raw = String(data.classification || '').toLowerCase();
  const classification = VALID.has(raw) ? raw : 'source';
  return {
    task: 'classify-page',
    page_path: pagePath,
    classification,
    confidence: typeof data.confidence === 'number' ? Math.max(0, Math.min(1, data.confidence)) : 0,
    reasoning: data.reasoning || '',
  };
}

File: agent/weft_agent/tasks/classify_page.py
"""classify-page — classify a page into source/entity/concept/synthesis
(port of llm/lib/tasks/classify-page.mjs).
"""
from ..runner import run_json_prompt

VALID = {"source", "entity", "concept", "synthesis"}


def run(kb_root, input=None, output_path=None):
    input = input or {}
    page_path = input.get("page_path") or ""
    res = run_json_prompt(kb_root, "classify-page", {
        "page_path": page_path,
        "title": input.get("title") or "",
        "body": input.get("body") or "",
    })
    data = res["data"]
    raw = str(data.get("classification") or "").lower()
    confidence = data.get("confidence")
    return {
        "task": "classify-page",
        "page_path": page_path,
        "classification": raw if raw in VALID else "source",
        "confidence": max(0.0, min(1.0, confidence)) if isinstance(confidence, (int, float)) else 0,
        "reasoning": data.get("reasoning") or "",
    }
